[18.0][FIX] account_reconcile_oca: Can't fetch record error to reconcile

hussain opened the pull request:
Before this fix using the account_reconcile_oca module in Odoo 18.0, opening the widget cause a loading module warning.

In the reconcile_data_widget.esm.js file where the widget wasn't being loaded correctly due to improper props definition.
This fix properly defines the widget props using standardFieldProps and moves the template definition to the class declaration,
ensuring the widget is loaded correctly and preventing the FetchRecordError after reconciliation.

This seems to be partially related #813 in a way it does not toast error message of deleted IDs not found in all cases, which I'm not sure why.


I signed OCA CLA.

victoralmau reviewed (changes requested):
Reviewing the error described at https://github.com/OCA/account-reconcile/issues/813 the error still occurs.

![ejemplo](https://github.com/user-attachments/assets/ff733521-0234-4127-9002-5f2014580295)

hussain commented:
I test it in runboat and actually it does work without error!


https://github.com/user-attachments/assets/717d83eb-87ef-49f3-9b93-222952d15871




but there is another issue: after reconcile/clear it does not allow selection to move to upper panel. Checking the issue.

victoralmau commented:
The error still occurs.

![ejemplo](https://github.com/user-attachments/assets/0b0bdd37-7821-4768-aa19-39e35651f545)

Important to select in the left panel the Test customer

![ejemplo](https://github.com/user-attachments/assets/4f0dd3ee-c14c-4176-b52b-610805ea2636)

hussain committed:
[FIX] account_reconcile_oca: Fix widget loading causing FetchRecordError

Before this fix using the account_reconcile_oca module in Odoo 18.0, clicking the 'Reconcile' button successfully reconciled
the entries but produced an error 'FetchRecordError: It seems the records with IDs 10 cannot be found. They might have been deleted.'.

The root cause was in the reconcile_data_widget.esm.js file where the widget wasn't being loaded correctly due to improper props definition.
This fix properly defines the widget props using standardFieldProps and moves the template definition to the class declaration,
ensuring the widget is loaded correctly and preventing the FetchRecordError after reconciliation.

hussain commented:
This is getting interesting!
There was initially a module loading issue which was fixed along same code change,I over sighted this and did not test before changing the other part!
Tested in runboat successfully for loading the widget, and for reconcile it seems less errors but there are cases  where it did toast the deleted IDs message but I cannot get the case identified and I'm not sure if it actually relate to this PR conceptually.

etobella commented:
I will invest today sometime in order to check why it happens.
